corehq/apps/api/es.py

```diff
@@ -226,8 +226,7 @@ def post(self, request, *args, **kwargs):
         More powerful ES querying using POST params.
         """
         try:
-            raw_post = request.body
-            raw_query = json.loads(raw_post)
+            raw_query = json.loads(request.body.decode('utf-8'))
```

Comment:
Should use `request.text` rather than `request.body.decode(...)`. The remote server may use a different encoding than UTF-8, and `request.text` would take that into account.

Reply:
Should ```request.text``` be used everywhere in the codebase that ```request.body.decode(...)``` currently is?

Reply:
Also, I don't see any examples of ```request.text``` in the code - could you point me to one?

Reply:
> Should ```request.text``` be used everywhere in the codebase that ```request.body.decode(...)``` currently is?

Yes, if the server sends a proper `Content-type` header (which I would assume is true in most cases). It may be that a particular server is known to indicate the wrong encoding (or none at all), in which case we may need to do something special like overriding `response.encoding` before calling `response.text`.  See also [docs on `response.text`](https://2.python-requests.org//en/master/user/quickstart/#response-content).

Here's an example:

https://github.com/dimagi/commcare-hq/blob/03499156b26de39121531b82e7ddbd5aa5405783/corehq/messaging/smsbackends/vertex/models.py#L84

Reply:
That's a response object, not a django request, no?

Reply:
Oh, heh, yes. I didn't realize this was a django request.